Amplify/Categories/DataStore/Model/Collection/List+Pagination.swift

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+extension List {
+
+    /// Check if there is subsequent data to retrieve. If true, the next page can be retrieved using
+    /// `getNextPage(completion:)`. Calling this will load the underlying elements from the data source if not yet
+    /// loaded before.
+    public func hasNextPage() -> Bool {
+        switch loadedState {
+        case .loaded:
+            return listProvider.hasNextPage()
+        case .notLoaded:
+            let result = listProvider.load()
+            switch result {
+            case .success(let elements):
+                self.elements = elements
+                return listProvider.hasNextPage()
+            case .failure(let coreError):
+                Amplify.log.error(error: coreError)
+                return false
+            }
+        }
+    }
+
+    /// Retrieve the next page as a new in-memory List object. Calling this will load the underlying elements from the
+    /// data source if not yet loaded before
+    public func getNextPage(completion: @escaping (Result<List<Element>, CoreError>) -> Void) {
+        switch loadedState {
+        case .loaded:
+            listProvider.getNextPage(completion: completion)
```

Comment:
In terms of platform parity, for Android in its current state-

- Android does not have lazy loading, and a list will always be created in a "loaded" state.
- getNextRequest returns nil when there is no next page. 

on IOS, when it is loaded, we delegate the call to the list provider to getNextPage. In the case where there is no next page, we return a failure. 

previous conversation: https://github.com/aws-amplify/amplify-ios/pull/990/files#r548594733